🐛 KakaoLogin has only one sessionCallback

greedy0110 opened the pull request:
# Problem
`KakaoLogin.login`: create new sessionCallback instance and add sessionCallback using `addCallback`

And It doesn't clear. So, multiple callbacks are registered in same time.

# ChangeLog
`KakaoLogin.login()`'s logic is changed.

## As-is
1. create new sessionCallback instance
2. add callback

## To-be
1. if it already has sessionCallback instance, then removeCallback
2. create new sessionCallback instance
3. add callback

greedy0110 committed:
🐛 KakaoLogin has only one sessionCallback

# Problem
`KakaoLogin.login`: create new sessionCallback instance and add sessionCallback using `addCallback`

And It doesn't clear. So, multiple callbacks are registered in one time.

# ChangeLog
`KakaoLogin.login()`'s logic is changed.

## As-is
1. create new sessionCallback instance
2. add callback

## To-be
1. if it already has sessionCallback instance, then removeCallback
2. create new sessionCallback instance
3. add callback

WindSekirun commented on library/src/main/java/com/github/windsekirun/rxsociallogin/kakao/KakaoLogin.kt:
```diff
@@ -58,6 +58,12 @@ class KakaoLogin constructor(activity: androidx.fragment.app.FragmentActivity) :
     override fun login() {
         checkSession()
+
+        if (sessionCallback != null) {
```
@greedy0110 LGTM, There may be some code improvements.

In L77, we have `removeCallback` logic in `onDestroy` methods, so it can be extract to one methods.

```
private fun removeSession() {
   if (sessionCallback != null) {
     Session.getCurrentSession().removeCallback(sessionCallback)
     sessionCallback = null
   }
}
```

However, it is not necessary, let me know.

greedy0110 committed:
🩹 extract removeSessionCallback to private function

comment: https://github.com/WindSekirun/RxSocialLogin/pull/60#discussion_r559070898

WindSekirun commented:
Released in 1.2.6-androidx, Thanks for contribution
